src/views/ShareCredentialSign/ShareCredentialSign.tsx

```diff
@@ -0,0 +1,167 @@
+import { browser } from 'webextension-polyfill-ts';
+import { useCallback, useState } from 'react';
+import {
+  RequestForAttestation,
+  Attestation,
+  Credential,
+} from '@kiltprotocol/core';
+import { DefaultResolver } from '@kiltprotocol/did';
+import {
+  IDidResolvedDetails,
+  ISubmitCredential,
+  MessageBodyType,
+} from '@kiltprotocol/types';
+
+import { getIdentityDidCrypto } from '../../utilities/identities/identities';
+import { usePopupData } from '../../utilities/popups/usePopupData';
+
+import { ShareInput } from '../../channels/shareChannel/types';
+import { shareChannel } from '../../channels/shareChannel/shareChannel';
+
+import {
+  PasswordField,
+  usePasswordField,
+} from '../../components/PasswordField/PasswordField';
+import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
+
+import { Selected } from '../ShareCredential/ShareCredential';
+
+import * as styles from './ShareCredentialSign.module.css';
+
+interface Props {
+  selected: Selected;
+  onCancel: () => void;
+}
+
+export function ShareCredentialSign({
+  selected,
+  onCancel,
+}: Props): JSX.Element | null {
+  const t = browser.i18n.getMessage;
+
+  const data = usePopupData<ShareInput>();
+
+  const { credentialRequest, verifierDid } = data;
+
+  const { challenge } = credentialRequest;
+
+  const { credential, identity, sharedProps } = selected;
+
+  const [error, setError] = useState<string | null>(null);
+
+  const passwordField = usePasswordField();
+
+  const handleSubmit = useCallback(
+    async (event) => {
+      event.preventDefault();
+
+      const { address } = identity;
+      const { password } = await passwordField.get(event);
+      const { encrypt, keystore, didDetails } = await getIdentityDidCrypto(
+        address,
+        password,
+      );
+
+      const request = RequestForAttestation.fromRequest(credential.request);
+      await request.signWithDid(keystore, didDetails, challenge);
+
+      const attestation = await Attestation.query(request.rootHash);
+
+      if (!attestation) {
+        setError(t('view_ShareCredentialSign_error'));
+        return;
+      }
+
+      const credentialInstance = Credential.fromCredential({
+        request,
+        attestation,
+      });
+
+      const presentation = await credentialInstance.createPresentation({
+        selectedAttributes: sharedProps,
+        signer: keystore,
+        claimerDid: didDetails,
+      });
+
+      const credentialsBody: ISubmitCredential = {
+        content: [presentation],
+        type: MessageBodyType.SUBMIT_CREDENTIAL,
+      };
+
+      const { details: verifierDidDetails } = (await DefaultResolver.resolveDoc(
+        verifierDid,
+      )) as IDidResolvedDetails;
+      if (!verifierDidDetails) {
+        throw new Error(`Cannot resolve the DID ${verifierDid}`);
+      }
```

Comment:
I think this will not be needed if you use `getDidDetails` instead of resolver above